src/super_gradients/training/utils/distributed_training_utils.py

```diff
@@ -412,3 +414,36 @@ def __init__(self):
             ">>> setup_device(multi_gpu=MultiGPUMode.DISTRIBUTED_DATA_PARALLEL, num_gpus=...)"
         )
         super().__init__(self.message)
+
+
+def maybe_all_reduce_tensor_average(device: str, tensor: torch.Tensor) -> torch.Tensor:
+    """
+    When in DDP- mean-reduces tensor from all devices.
+    When not in DDP - returns the input tensor.
+
+    :param device:
+    :param tensor:
+    :return:
+    """
+    if is_distributed():
+        tensor = distributed_all_reduce_tensor_average(tensor=tensor.to(device), n=torch.distributed.get_world_size())
+    return tensor
+
+
+def maybe_all_gather_np_images(image: np.ndarray) -> np.ndarray:
```

Comment:
I believe this method's signature is a bit confusing. 
From my perspective it should be a List[np.ndarray] and not single np.ndarray. 
And the implementation should be like this:

```
    if is_distributed():
        rank = get_rank()
        output_container = [None for _ in range(get_world_size())]
        all_gather_object(output_container, image)
        return output_container
    return [image]
```

Why:

1) Current implementation behaves differently for master/non-master nodes. Which may be ok for current use case, but may fail in other scenarios.

2) It concatenates arrays along first dim, while this may be undesirable/impossible. Example - I want to gather arrays of predicted boxes which may have different length along axis=0. In this case the code would not work.

Reply:
Why does it behave differently ?
instead of returning np.array sized BCHW it returns B_sumranksCHW, the concatination will work for different sizes across dim0, so I dont get your point.

That is not the purpose of this method, the name of it and docs are pretty clear on that.